glean-core/src/upload/directory.rs

```diff
@@ -183,50 +221,39 @@ impl PingDirectoryManager {
                         self.delete_file(file_name);
                         return None;
                     }
-                    // In case we can't process the file we just ignore it.
                     if let Some(data) = self.process_file(file_name) {
-                        // Get the modified date of the file, which will later be used
-                        // for sorting the resulting vector.
-                        let modified_date = fs::metadata(&path).and_then(|data| data.modified());
-                        return Some((modified_date, data));
+                        // This may panic in case the path does not exist
+                        // or the user does not have permissions to read the file.
+                        // Both are extremely unlike so we can panic! here if that happens.
```

Comment:
Uhm ... not so sure about that, I've seen to many broken things with files :D 
If this ever _does_ panic it will take down the whole app.
But in case of an error we can just ignore it, can't we?

Reply:
We can't really ignore, we would have to not enqueue that file all together, or I guess set a default size for it or something of the sort.

My point in this being very unlikely is that we read the file in the lines right above this one in this same closure. If the path didn't exist / user does not have permissions we wouldn't really get to this line of the function.

For good measure I can change this to just ignore any failures though.

Reply:
Hm, in that case if we actually panic and crash we have a better indicator than not seeing it ever.